src/Phinx/Db/Adapter/MysqlAdapter.php

```diff
@@ -298,6 +298,103 @@ public function createTable(Table $table, array $columns = [], array $indexes =
         $this->execute($sql);
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    protected function getChangeTableInstructions(Table $table, array $newOptions)
+    {
+        $instructions = new AlterInstructions();
+
+        // Drop the existing primary key
+        $primaryKey = $this->getPrimaryKey($table->getName());
+        if ((isset($newOptions['id']) || isset($newOptions['primary_key']))
+            && !empty($primaryKey['columns'])) {
+            foreach ($primaryKey['columns'] as $column) {
+                $sql = sprintf(
+                    'MODIFY %s INT NOT NULL',
+                    $this->quoteColumnName($column)
+                );
+                $instructions->addAlter($sql);
+            }
+            $instructions->addAlter('DROP PRIMARY KEY');
+        }
+
+        // Set the default primary key and add associated column
+        if (isset($newOptions['id']) && $newOptions['id'] !== false) {
+            if ($newOptions['id'] === true) {
+                $newOptions['primary_key'] = 'id';
+            } elseif (is_string($newOptions['id'])) {
+                // Handle id => "field_name" to support AUTO_INCREMENT
+                $newOptions['primary_key'] = $newOptions['id'];
+            } else {
+                throw new \InvalidArgumentException(sprintf(
+                    "Invalid value for option 'id': %s",
+                    json_encode($newOptions['id'])
+                ));
+            }
+
+            if ($this->hasColumn($table->getName(), $newOptions['primary_key'])) {
+                throw new \RuntimeException(sprintf(
+                    "Tried to create primary key column %s for table %s, but that column already exists",
+                    $this->quoteColumnName($newOptions['primary_key']),
+                    $this->quoteTableName($table->getName())
+                ));
+            }
+
+            $column = new Column();
+            $column
+                ->setName($newOptions['primary_key'])
+                ->setType('integer')
```

Comment:
How do you know the type is integer?

Reply:
This makes me think that perhaps changing the primary key should provide a column object instead